### PR TITLE
Add robots.txt

### PR DESCRIPTION
**Pull request by thomasdax98**
Looks like this:

```
User-Agent: *
Allow: /

Sitemap: http://localhost:3000/sitemap.xml
```

---

## PR Checklist

-   [x] Link to the respective task if one exists: COM-988
-   [x] Provide screenshots/screencasts if the change contains visual changes

<details>
    <summary>Screenshots/screencasts</summary>
    

Bildschirmfoto 2024-07-31 um 09 30 09

Bildschirmfoto 2024-07-31 um 09 30 05

</details>

### CODE REVIEW

File: site/src/middleware.ts [hunk 1]
```diff
@@ -87,7 +87,7 @@ export const config = {
          * - favicon.ico, favicon.svg, favicon.png
          * - manifest.json
          */
-        "/((?!api|_next/static|_next/image|favicon.ico|favicon.svg|favicon.png|manifest.json|assets).*)",
+        "/((?!api|_next/static|_next/image|favicon.ico|favicon.svg|favicon.png|manifest.json|assets|robots.txt).*)",
```
Comment (johnnyomair) [inline site/src/middleware.ts:90]: You'd need to add it to the comment (or remove the comment). We could also consider adding a `const publicAssets = ["favicon.ico", "favicon.svg", ...]` array for the public assets.

File: site/src/app/robots.ts [hunk 2]
```diff
@@ -1,11 +1,19 @@
...
+        sitemaps.push(`${siteConfig.url}/sitemap.xml`);
+    }
+
     return {
         rules: {
             userAgent: "*",
             allow: "/",
         },
-        sitemap: `${process.env.SITE_URL}/sitemap.xml`, // TODO support multiple site domains
+        sitemap: sitemaps,
```
Comment (johnnyomair) [inline site/src/app/robots.ts:17]: I don't think that's correct. It should only include the sitemap of the current site config.

Comment (thomasdax98) [reply site/src/app/robots.ts]: [fad7eac](https://github.com/vivid-planet/comet-starter/pull/318/commits/fad7eacfe088e457c94be125c4ecec1874c16ee3)

File: site/src/middleware.ts (hunk 1 above)
Comment (thomasdax98) [reply site/src/middleware.ts:91]: [32ccdab](https://github.com/vivid-planet/comet-starter/pull/318/commits/32ccdabaa7c85d7b0376dd80c0d501421008ecaa)

**Review by dkarnutsch — DISMISSED** [review]

**Review by johnnyomair — APPROVED** [review]

File: site/src/app/robots.ts [hunk 3]
```diff
@@ -1,11 +1,15 @@
+import { getHost, getSiteConfigForDomain } from "@src/config";
 import { MetadataRoute } from "next";
+import { headers } from "next/headers";
 
 export default async function robots(): Promise<MetadataRoute.Robots> {
+    const siteConfig = getSiteConfigForDomain(getHost(headers()));
```
Comment (nsams) [inline site/src/app/robots.ts:6]: is that correct? getHost returns `"main.localhost:3000"` but getSiteConfigForDomain expects `"main"`

**Review by nsams — APPROVED** [review]

**Review by nsams — APPROVED** [review]

**Pull request merged**